client/layout/index.jsx

```diff
@@ -300,7 +303,7 @@ class Layout extends Component {
 						{ this.props.secondary }
 					</div>
 					<div id="primary" className="layout__primary">
-						{ this.props.primary }
+						{ this.props.signupIsInitializing ? '' : this.props.primary }
```

Comment:
Are there any negative externalities from causing a full re-render here?

Reply:
Not due to re-rendering. Rendering shouldn't have any side effects so re-rendering is always safe.

But it's possible that mounting could cause issues. This change means that the `componentDidMount` methods and `useEffect` callbacks won't be called until after `indicateEndSignupStateChanges`. I guess theoretically it's possible that some of this code depends on things like the site ID being `null` at the time of mount.
I don't think it'll cause issues, but it is possible ...